### ISSUE TITLE
Storing of Template is extremely slow

### ISSUE THREAD

**Issue by ppanopticon** [dres-dev/DRES#407]
Storing of task templates (update) is to slow. The logic in `UpdateEvaluationTemplateHandler` should therefore be revisited.

**Comment by lucaro** [comment]
I can confirm that it was rather quick before adding the whole LSC dataset as a collection. So the processing time depends on the total database size. Maybe there's an option to add some custom indices that would help?

**Comment by sauterl** [comment]
There was a discussion on using internal (xodus provided) IDs, which would require a little refactoring effort

**Comment by ppanopticon** [comment]
I fixed a range of (potential) performance issues.

From my analysis I could deduce that most time was spent in the logic that updates the existing tasks and, more precisely, that performs the lookup for existing media items. Here it seems that whatever index Xodus uses fails if the operand is null despite the attribute value being non-nullable. It probably leads to a full-scan of all the 750'000 items.

With the changes I've applied, the time it takes to persist tasks is now in the millisecond range (down from 7-8 seconds). I imagine that there might be some additional potential if we used the official xdId instead of our artificially generated UUIDs. However, this might be a larger undertaking.
